src/pgduckdb_hooks.cpp

```diff
@@ -197,7 +216,7 @@ DuckdbPlannerHook_Cpp(Query *parse, const char *query_string, int cursor_options
 			IsAllowedStatement(parse, true);
 
 			return DuckdbPlanNode(parse, query_string, cursor_options, bound_params, true);
-		} else if (duckdb_force_execution && IsAllowedStatement(parse)) {
+		} else if (duckdb_force_execution && IsAllowedStatement(parse) && ContainsFromClause(parse)) {
```

Comment:
You want to move `ContainsFromClause(parse)` before `IsAllowedStatement(parse)` no? o/w we could hit the same issue calling `ContainsCatalogTable` w/ a `nullptr` no?

Reply:
It's fine to call ContainsCatalogTable and ContainsPartitionedTable with a nullptr. They both call `foreach_node` on that pointer, and the nullptr is a valid (and I think the only valid) representation of an empty list.